### PR TITLE
feat(authentication-service): provide default implementation for azure ad

### LINKED ISSUE #931: Provide default implementation for OAUTH via Azure-Ad
**Describe the bug**
Default implementation for Azure-Ad is missing

### PR DESCRIPTION
**Pull request by yeshamavani**
gh-931

## Description

Provide default implementation for Azure Ad

Fixes #931 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/authentication-service/.env.defaults [hunk 1]
```diff
@@ -27,4 +27,24 @@ REDIS_URL=
...
+AZURE_AUTH_REDIRECT_URL=url
+AZURE_AUTH_CLIENT_SECRET=client_secret
+AZURE_AUTH_ALLOW_HTTP_REDIRECT=true
+AZURE_AUTH_COOKIE_INSTEAD_SESSION=true
+AZURE_AUTH_PASS_REQ_CALLBACK=
+AZURE_AUTH_VALIDATE_ISSUER=
+
+# key is 32 bit
+
+AZURE_AUTH_COOKIE_KEY=gthnbftrgabdukdineikeccnauyftrdf
```
Comment (akshatdubeysf) [inline services/authentication-service/.env.defaults:46]: Remove the default cookie, this should always come from user

File: services/authentication-service/.env.defaults [hunk 2]
```diff
@@ -27,4 +27,24 @@ REDIS_URL=
...
+
+#iv is 12 bit
+
+AZURE_AUTH_COOKIE_IV=utuyjhlijiug
```
Comment (akshatdubeysf) [inline services/authentication-service/.env.defaults:50]: same

File: services/authentication-service/src/component.ts [hunk 3]
```diff
@@ -93,6 +102,14 @@ export class AuthenticationServiceComponent implements Component {
     // Mount core component
     this.application.component(CoreComponent);
 
+    const expressMiddlewares =
+      this.application.getSync(SFCoreBindings.EXPRESS_MIDDLEWARES) ?? [];
+    expressMiddlewares.push(cookieParser());
```
Comment (akshatdubeysf) [inline services/authentication-service/src/component.ts:107]: Add these on a condition and note it in the README

File: services/authentication-service/src/modules/auth/azure-login.controller.ts [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+        content: {
+          [CONTENT_TYPE.JSON]: {
+            schema: {[X_TS_TYPE]: TokenResponse},
+          },
+        },
+      },
+    },
+  })
+  async azureCallback(
+    @param.query.string('code') code: string, //NOSONAR
```
Comment (akshatdubeysf) [inline services/authentication-service/src/modules/auth/azure-login.controller.ts:191]: why no sonar?

**Review by akshatdubeysf — CHANGES_REQUESTED** [review]

File: services/authentication-service/src/modules/auth/azure-login.controller.ts (hunk 4 above)
Comment (yeshamavani) [reply services/authentication-service/src/modules/auth/azure-login.controller.ts:191]: sonar gives error or unused var

File: services/authentication-service/.env.defaults (hunk 1 above)
Comment (yeshamavani) [reply services/authentication-service/.env.defaults:46]: okay

File: services/authentication-service/.env.defaults (hunk 2 above)
Comment (yeshamavani) [reply services/authentication-service/.env.defaults:50]: done

**Pull request merged**